src/SpendPermissionManager.sol

```diff
@@ -259,22 +272,30 @@ contract SpendPermissionManager is EIP712 {
         }
 
         // loop through each spend permission in the batch and approve it
+        bool allApproved = true;
         uint256 batchLen = spendPermissionBatch.permissions.length;
         for (uint256 i; i < batchLen; i++) {
-            _approve(
-                SpendPermission({
-                    account: spendPermissionBatch.account,
-                    spender: spendPermissionBatch.permissions[i].spender,
-                    token: spendPermissionBatch.permissions[i].token,
-                    allowance: spendPermissionBatch.permissions[i].allowance,
-                    period: spendPermissionBatch.period,
-                    start: spendPermissionBatch.start,
-                    end: spendPermissionBatch.end,
-                    salt: spendPermissionBatch.permissions[i].salt,
-                    extraData: spendPermissionBatch.permissions[i].extraData
-                })
-            );
+            // approve each spend permission in the batch, capturing a false return to surface if any return false (are
+            // already revoked)
+            if (
+                !_approve(
+                    SpendPermission({
+                        account: spendPermissionBatch.account,
+                        spender: spendPermissionBatch.permissions[i].spender,
+                        token: spendPermissionBatch.permissions[i].token,
+                        allowance: spendPermissionBatch.permissions[i].allowance,
+                        period: spendPermissionBatch.period,
+                        start: spendPermissionBatch.start,
+                        end: spendPermissionBatch.end,
+                        salt: spendPermissionBatch.permissions[i].salt,
+                        extraData: spendPermissionBatch.permissions[i].extraData
+                    })
+                )
+            ) {
+                allApproved = false;
```

Comment:
as devx / user experience note here, if one of the approvals in the batch returns false (meaning it's been revoked), we don't have a way of knowing which permission in the batch was actually revoked.

one way I can think of handling this case is to emit an event after we flip the `allApproved = false` here on line 295. However, that would require us to make an extra call to hash the `SpendPermission` and then emit the hash, which might not be ideal. Not sure if it's a big enough issue to try and fix, but think its worth calling out